lib/export-scala-regmap.js

```diff
@@ -170,12 +191,41 @@ const getAddressBlockNames = addressBlock => {
     ioBundle: `${addressBlock.name}AddressBlockBundle`,
     regMapRegister: 'register',
     regMapResetWire: 'resetValue',
+    paramsName: 'componentParams',
     tlRegMap: `${addressBlock.name}TLRegMap`,
     axi4RegMap: `${addressBlock.name}AXI4RegMap`
   };
 };
 
-const generateBodyBase = (names, addressBlock, deviceNameString, compatString) => {
+const padFieldsMethod = endent`
+  private def padFields(fields: (Int, RegField)*) = {
+    var previousOffset = 0
+    var previousField: Option[RegField] = None
+
+    fields.flatMap { case (fieldOffset, field) =>
+      val padWidth = fieldOffset - previousOffset
+      require(padWidth >= 0,
+        previousField.map(f => s"register fields at $previousOffset and $fieldOffset are overlapping")
+          .getOrElse(s"register field $field has a negative offset"))
```

Comment:
Nit: I think this is more readable as

```scala
if (previousField.isDefined) "first string" else "second string"
```

since you aren't doing anything with the actual previous field anyway.

Double nit: Capitalize error messages, especially ones that are basically full sentences.